src/query/ast/src/parser/expr.rs

```diff
@@ -1126,9 +1126,7 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
     let ty_date = value(TypeName::Date, rule! { DATE });
     let ty_datetime = map(
         rule! { (DATETIME | TIMESTAMP) ~ ( "(" ~ #literal_u64 ~ ")" )? },
```

Comment:
Also remove `#literal_u64` ?

Reply:
Some SQL dialects may need this, I prefer just keep it now.